tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -80,10 +90,16 @@ def test_slurm(
         scaledown_idletime=scaledown_idletime,
         gpu_instance_type=gpu_instance_type,
         compute_node_bootstrap_timeout=compute_node_bootstrap_timeout,
+        use_login_node=use_login_node,
     )
     cluster = clusters_factory(cluster_config, upper_case_cluster_name=True)
-    remote_command_executor = RemoteCommandExecutor(cluster)
-    clustermgtd_conf_path = _retrieve_clustermgtd_conf_path(remote_command_executor)
+    remote_command_executor = RemoteCommandExecutor(cluster, use_login_node=use_login_node)
+    remote_command_executor_for_head_node = None
+    if use_login_node:
+        remote_command_executor_for_head_node = RemoteCommandExecutor(cluster)
+    clustermgtd_conf_path = _retrieve_clustermgtd_conf_path(
+        remote_command_executor if not use_login_node else remote_command_executor_for_head_node
+    )
```

Comment:
Since 
 * `clustermgtd_conf_path` is only used in `_test_compute_node_bootstrap_timeout(` below in the code 
 
and 
* `_test_compute_node_bootstrap_timeout(`
*  `assert_no_errors_in_logs(` 

must be always run in the HeadNode I propose to move the logic that must be run in the head node below and keep it together. Also to highlight the fact that those tests MUST be run on the head node I'd create a dedicated command executor

```
head_node_command_executor = RemoteCommandExecutor(cluster)
...
 assert_no_errors_in_logs(head_node_command_executor)
 ...
_test_compute_node_bootstrap_timeout(..., head_node_command_executor)
 ```

while in the first part of the test always use a variable called `remote_command_executor` (that may be initialized differently depending on the `use_login_node` parameter) to make it more clear that this section can be run without changes both in a LoginNode or in the HeadNode.

Reply:
Makes sense! Thank you Nicola!